code/datums/elements/openspace_item_click_handler.dm

```diff
@@ -8,22 +8,19 @@
 	. = ..()
 	if(!isitem(target))
 		return ELEMENT_INCOMPATIBLE
-	RegisterSignal(target, COMSIG_ITEM_INTERACTING_WITH_ATOM, PROC_REF(divert_interaction))
+	RegisterSignal(target, COMSIG_RANGED_ITEM_INTERACTING_WITH_ATOM, PROC_REF(divert_interaction))
 
 /datum/element/openspace_item_click_handler/Detach(datum/source)
-	UnregisterSignal(source, COMSIG_ITEM_INTERACTING_WITH_ATOM)
+	UnregisterSignal(source, COMSIG_RANGED_ITEM_INTERACTING_WITH_ATOM)
 	return ..()
 
 //Invokes the proctype with a turf above as target.
-/datum/element/openspace_item_click_handler/proc/divert_interaction(obj/item/source, mob/user, atom/target, click_parameters)
+/datum/element/openspace_item_click_handler/proc/divert_interaction(obj/item/source, mob/user, atom/target, list/modifiers)
 	SIGNAL_HANDLER
 	if((target.z == 0) || (user.z == 0) || target.z == user.z)
 		return NONE
-	var/turf/checked_turf = get_turf(target)
-	while(!isnull(checked_turf))
-		checked_turf = GET_TURF_ABOVE(checked_turf)
-		if(checked_turf?.z == user.z && user.CanReach(checked_turf, source))
-			INVOKE_ASYNC(source, TYPE_PROC_REF(/obj/item, handle_openspace_click), checked_turf, user, click_parameters)
-			return ITEM_INTERACT_BLOCKING
-
+	var/turf/target_turf = parse_caught_click_modifiers(modifiers, get_turf(user.client ? user.client.eye : user), user.client)
```

Comment:
Why is this using `user.client.eye` now? 

(Either way it can be shortened to `user.client?.eye || user`

Reply:
Took the usage from here
https://github.com/tgstation/tgstation/blob/bf7cea7c9de5de3b7e97b2f43ab9b0d8d47a5ab1/code/_onclick/click.dm#L433
It needs to know where the user is actually looking in order to figure out what turf they're aiming at